### ISSUE TITLE
Weights for the CLIP image encoder

### ISSUE THREAD

**Issue by juexZZ** [qihao067/CrossFlow#14]
Hi Qihao,

Thank you for open-sourcing this great work.

I was going over the code and want to confirm some of the details:

### 1. the weight loading 
I see that the pre-trained weights for the open_clip (used for encoding the image for the image-text CLIP loss) model are not loaded:

https://github.com/qihao067/CrossFlow/blob/8493373c6e5c53b2b02f38586a4cf8e45e287c7d/libs/model/dimr_t2i.py#L358

https://github.com/qihao067/CrossFlow/blob/8493373c6e5c53b2b02f38586a4cf8e45e287c7d/libs/model/dit_t2i.py#L175

Is this done on purpose?  I went further and compared the params with the original model:
```python
print("crossflow")
print(next(nnet.open_clip.visual.parameters()))
print("reload from webli")
import open_clip
clip_webli, ptrain, pval = open_clip.create_model_and_transforms('ViT-L-16-SigLIP-256', pretrained='webli')
print(next(clip_webli.visual.parameters()))
```
And I do get different weights:
```
crossflow
Parameter containing:
tensor([[[ 0.0541,  0.0041,  0.0805,  ..., -0.0062,  0.0335, -0.0036],
         [-0.0309, -0.0085, -0.0217,  ..., -0.0086, -0.0345,  0.0147],
         [-0.0243, -0.0107, -0.0015,  ..., -0.0054, -0.0555,  0.0066],
         ...,
         [-0.0110, -0.0048, -0.0216,  ...,  0.0003,  0.0271, -0.0035],
         [ 0.0275, -0.0071, -0.0505,  ..., -0.0176,  0.1224,  0.0138],
         [ 0.0409,  0.0084, -0.0712,  ..., -0.0040,  0.1701,  0.0276]]],
       device='cuda:0', requires_grad=True)
reload from webli
2025-06-18 22:06:56,701 - factory.py - Loaded ViT-L-16-SigLIP-256 model config.
2025-06-18 22:07:12,323 - factory.py - Loading pretrained ViT-L-16-SigLIP-256 weights (webli).
Parameter containing:
tensor([[[ 0.0361, -0.0199, -0.0707,  ...,  0.0657,  0.0523,  0.0110],
         [ 0.0325, -0.0098,  0.0430,  ...,  0.0719,  0.1004,  0.0258],
         [ 0.0059, -0.0205,  0.0527,  ...,  0.0348,  0.0846,  0.0054],
         ...,
         [ 0.0384, -0.0010, -0.0814,  ...,  0.1298, -0.0682,  0.0212],
         [-0.0019, -0.0310, -0.0855,  ...,  0.1177,  0.2483,  0.0048],
         [ 0.1717, -0.0178, -0.1623,  ...,  0.1415,  0.1389,  0.0182]]],
       requires_grad=True)
```

So is it correct that you trained end-to-end from scratch for the image encoder in the CLIP loss?

### 2. model choice
I also notice that for the CLIP text embedder, you use "openai/clip-vit-large-patch14":
https://github.com/qihao067/CrossFlow/blob/8493373c6e5c53b2b02f38586a4cf8e45e287c7d/libs/clip.py#L16
And for the image CLIP side, you choose ViT-L-16-SigLIP-256. So I am curious to know the reason of this mismatch.

Thank you!

Best,

**Comment by qihao067** [comment]
Hello @juexZZ 

Thank you for the thoughtful questions and for taking the time to carefully go through the code.

1. Weight loading
Yes, you're correct. In my original implementation, I built the image encoder myself. For this reimplementation, I used the open_clip model definition for convenience but initialized it with pretrained=None, meaning we did not load any pretrained weights. We train the image encoder from scratch as part of our full pipeline. In fact, other image encoders should work as well. this setup was chosen for simplicity for a quick release.

2. Model choice
For the text encoder, we use "openai/clip-vit-large-patch14", which I think is a common and well-established choice in prior work. As for the image encoder, other alternatives could likely work. What we used here is just for simplicity for a quick release. Due to resource and time constraints, we didn’t actiually have the chance to conduct ablations on different model architectures. Also, I wouldn’t exactly call it a mismatch as in our framework, the image encoder is paired with the variational text encoder we proposed in the paper, not the standard CLIP text encoder.

Hope this answers your questions. Please let me know if anything is unclear or if you’d like to discuss further.

Best,
Qihao

**Comment by juexZZ** [comment]
Thank you for your answer!
